src/services/projectService.test.ts

```diff
@@ -146,15 +148,106 @@ describe("Project Service", () => {
         expect(projectSerivce.isDuplicate(testProject, projectList)).toEqual(true);
     });
 
-    it("deletes all asset metadata files when project is deleted", async () => {
-        const assets = MockFactory.createTestAssets(10);
+    function populateProjectAssets(project?: IProject, assetCount = 10) {
+        if (!project) {
+            project = MockFactory.createTestProject();
+        }
+        const assets = MockFactory.createTestAssets(assetCount);
         assets.forEach((asset) => {
             asset.state = AssetState.Tagged;
         });
 
-        testProject.assets = _.keyBy(assets, (asset) => asset.id);
+        project.assets = _.keyBy(assets, (asset) => asset.id);
+        return project;
+    }
+
+    it("deletes all asset metadata files when project is deleted", async () => {
+        const assetCount = 10;
+        populateProjectAssets(testProject);
 
         await projectSerivce.delete(testProject);
-        expect(storageProviderMock.deleteFile.mock.calls).toHaveLength(assets.length + 1);
+        expect(storageProviderMock.deleteFile.mock.calls).toHaveLength(assetCount + 1);
+    });
+
+    it("Deletes tag from all assets within project", async () => {
+        const tag1 = "tag1";
+        const tag2 = "tag2";
+        const region = MockFactory.createTestRegion(undefined, [tag1, tag2]);
+        const asset: IAsset = {
+            ...MockFactory.createTestAsset("1"),
+            state: AssetState.Tagged,
+        };
+        const assetMetadata = MockFactory.createTestAssetMetadata(asset, [region]);
+        AssetService.prototype.getAssetMetadata = jest.fn((asset: IAsset) => Promise.resolve(assetMetadata));
+
+        const saveMetadata = jest.fn();
+        AssetService.prototype.save = saveMetadata;
+
+        const expectedAssetMetadata: IAssetMetadata = {
+            ...MockFactory.createTestAssetMetadata(
+                asset,
+                [
+                    {
+                        ...region,
+                        tags: [tag2],
+                    },
+                ],
+            ),
+
+        };
+        const project = populateProjectAssets();
+        await projectSerivce.deleteTag(project, tag1, assetMetadata);
+        expect(saveMetadata).toBeCalledWith(expectedAssetMetadata);
     });
+
+    it("Deletes any empty regions after deleting only tag from region", async () => {
```

Comment:
I think this is very aggressive and we should alert user warning them if they're deleting the only tag of a region